Added first draft of SSI with CNIs (#31666)
DataDog/documentation: content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/kubernetes.md

For more details on changing your container registry, see [Changing Your Container Registry][33].

> ### Using a Container Network Interface on EKS

Reviewer: ```suggestion
### Use a Container Network Interface on EKS
```

> When using a CNI like Calico, the control plane nodes will not be able to initiate network connections to Datadog's Admission Controller with a "Address is not allowed" error".

Reviewer: ```suggestion
When using a CNI like Calico, the control plane nodes are not able to initiate network connections to Datadog's Admission Controller and report an "Address is not allowed" error.
```

> If you sant Datadog' Single Step instrumentation to work, modify Datadog's Cluster Agent with the `useHostNetwork: true` parameter.

Reviewer: ```suggestion
To use Single Step instrumentation, modify Datadog's Cluster Agent with the `useHostNetwork: true` parameter.
```
